docs/reference/modules/cross-cluster-search.asciidoc

```diff
@@ -0,0 +1,112 @@
+
+[source,yaml]
+--------------------------------
+search:
+    remote:
+        seeds:
+            cluster_one: 127.0.0.1:9300 <1>
+            cluster_two: 127.0.0.1:9301 <1>
+
+--------------------------------
+<1> `cluster_one` and `cluster_two` are arbitrary names representing the connection to each cluster. These names are subsequently used to distinguish between local and remote indices.
+
+[float]
+=== Using cross cluster search
+
+To search the `twitter` index on remote cluster `cluster_1` the index name must be prefixed with the cluster name
+separated by a `:` character:
+
+[source,js]
+--------------------------------------------------
+POST /cluster_one:twitter/tweet/_search
+{
+    "match_all": {}
+}
+--------------------------------------------------
+
+In contrast to the `tribe` feature cross cluster search can also search indices with the same name on different clusters:
+
+
+[source,js]
+--------------------------------------------------
+POST /cluster_one:twitter,twitter/tweet/_search
+{
+    "match_all": {}
+}
+--------------------------------------------------
+
+Search results are disambiguated the same way as the indices are disambiguated in the request. Even if index names are
+identical these indices will be treated as different indices when results are merged. All results retrieved from a remote index
+will be prefixed with their remote cluster name:
+
+[source,js]
+--------------------------------------------------
+ {
+  "took" : 89,
+  "timed_out" : false,
+  "_shards" : {
+    "total" : 10,
+    "successful" : 10,
+    "failed" : 0
+  },
+  "hits" : {
+    "total" : 2,
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "cluster_one:twitter",
+        "_type" : "tweet",
+        "_id" : "1",
+        "_score" : 1.0,
+        "_source" : {
+          "user" : "kimchy",
+          "postDate" : "2009-11-15T14:12:12",
+          "message" : "trying out Elasticsearch"
+        }
+      },
+      {
+        "_index" : "twitter",
+        "_type" : "tweet",
+        "_id" : "1",
+        "_score" : 1.0,
+        "_source" : {
+          "user" : "kimchy",
+          "postDate" : "2009-11-15T14:12:12",
+          "message" : "trying out Elasticsearch"
+        }
+      }
+    ]
+  }
+}
+--------------------------------------------------
+
+[float]
+=== Cross cluster search settings
+
+* `search.remote.connections_per_cluster` - the number of nodes to connect to per remote cluster. The default is `3`
+* `search.remote.initial_connect_timeout` - the time to wait for remote connections to be established when the node starts. The default is `30s`.
+* `search.remote.node_attribute` - a node attribute to filter out nodes that are eligible as a gateway node in the remote cluster.
+For instance a node can have a node attribute `node.attr.gateway: true`  such that only nodes with this attribute
+will be connected to if `search.remote.node_attribute` is set to `gateway`
```

Comment:
shall we add some info on versions compatibility. 5.3 + right? the proxy actions are not exposed in previous versions for instance, or are there conditions in which we expect this to work with previous 5.x versions?